book/embeds.md

```diff
@@ -1341,6 +1341,57 @@ class FrameAccessibilityNode(AccessibilityNode):
         return node
 ```
 
+Hit testing should now work, but the bounds of the hovered node when drawn
+to the screen are still wrong. For that, we'll need a method that returns
+the absolute screen rect of an `AccessibilityNode`. And that method in turn
+needs parent pointers so we can walk up the tree, so let's add that first:
+
+``` {.python}
+class AccessibilityNode:
+    def __init__(self, node, parent = None):
+        # ...
+        self.parent = parent
+
+    def build_internal(self, child_node):
+        # ...
+            child = FrameAccessibilityNode(child_node, self)
+        else:
+            child = AccessibilityNode(child_node, self)
+```
+
+And now the method to map to absolute coordinates:
+
+``` {.python}
+class AccessibilityNode:
+    def absolute_bounds(self):
+        rect = skia.Rect.MakeXYWH(
+            self.bounds.x(), self.bounds.y(),
+            self.bounds.width(), self.bounds.height())
+        obj = self
+        while obj:
+            obj.map_to_parent(rect)
+            obj = obj.parent
+        return rect
+```
+
+This method does nothing until we've defined `map_to_parent` of course, and
+that does nothing for most accessibility nodes (because they are in the same
+coordinate space of their parent):
```

Comment:
```suggestion
This method depends calls `map_to_parent` to adjust the bounds. For
 most accessibility nodes we don't need to do anything, because they are in the same
coordinate space as their parent:
```